AK/StdShim.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2025, Leon Albrecht <leon.a@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/StdLibExtraDetails.h>
+
+// Note:
+// These functions we might need to implement to get some stuff working in constexpr contexts,
+// Or because they'd are visible in the standard library and we need to provide them,
+// So we need to include the official implementations and poison the rest of std in lagom builds
+// for serenity builds we are the standard lib so we can implement them ourselves
+// ALSO NOTE for move/forward: These have to be in the "std" namespace since some compilers and static analyzers rely on it.
+
+// Note: Named in a way that will sort this after other headers that might need to include stl headers
+
+#ifdef AK_OS_SERENITY
```

Comment:
Oh actually, it just occurred to me that somebody might want to use `-DAK_DONT_REPLACE_STD` on Serenity as well, so this condition should be `#if defined(AK_OS_SERENITY) && !defined(AK_DONT_REPLACE_STD)`

Reply:
... although, we potentially change ABI by doing this. So, to be on a safe side, that somebody should somehow get a copy of system libraries built with `-DAK_DONT_REPLACE_STD`.

Maybe it's best to just guard this by `defined(KERNEL)`? @ADKaster, wdyt?

Reply:
> Maybe it's best to just guard this by defined(KERNEL)?

And with this, we can get rid of `AK_DONT_REPLACE_STD`. ~~Preprocessor defines parity with ladybird, anyone?~~

Reply:
As discussed, not with a lot more leaky stl includes, and it's hard to find a common point to poison std, as poisoning std also breaks any future includes of std